test/api/test_module_construct.cpp

```diff
@@ -1,16 +1,56 @@
+#include <numeric>
 #include <migraphx/migraphx.h>
 #include <migraphx/migraphx.hpp>
 #include "test.hpp"
 
+void run_test_on_gpu(const migraphx::program p)
+{
+    // make a copy of the program
+    migraphx_program_t p_ref_c;
+    migraphx_program_create(&p_ref_c);
+    migraphx_program_assign_to(p_ref_c, p.get_handle_ptr());
+    auto p_ref = migraphx::program(p_ref_c, migraphx::own{});
+    // compile on GPU
+    migraphx::compile_options options;
+    options.set_offload_copy();
+    p.compile(migraphx::target("gpu"), options);
+    // set parameters_map
+    migraphx::program_parameters pp;
+    auto param_shapes = p.get_parameter_shapes();
+    for(auto&& name : param_shapes.names())
+    {
+        pp.add(name, migraphx::argument::generate(param_shapes[name]));
```

Comment:
We cant use random numbers. Just fill it the same value. We aren't trying to do numeric validation here, just to need to check that the program ran as expected.

Reply:
Fixed.